libass/ass_parse.c

```diff
@@ -194,11 +195,28 @@ inline void change_alpha(uint32_t *var, int32_t new, double pwr)
  * \param a first value
  * \param b second value
  * \return result of multiplication
- * Parameters and result are limited by 0xFF.
+ * At least one of the parameters must be less than or equal to 0xFF.
+ * The result is less than or equal to max(a, b, 0xFF).
  */
 inline uint32_t mult_alpha(uint32_t a, uint32_t b)
 {
-    return 0xFF - (0xFF - a) * (0xFF - b) / 0xFF;
+    return a - ((uint64_t) a * b + 0x7F) / 0xFF + b;
```

Comment:
Forgot to say: I realize that this expression isn’t exactly immediately obvious (especially how it avoids overflow and how it’s dangerous to reorder the operations), so I’m open to suggestions for improving it. This is the simplest expression I’ve found though; but maybe simpler is not strictly better in this case.

Reply:
Fun comment here re: rounding: https://github.com/Cyberbeing/xy-VSFilter/blob/ff77719517bc0c1144a75c7b2c1e6268d582614a/src/subtitles/RTS.cpp#L1257

Pretty sure this rounding itself is otherwise correct, given that vsfilter's using this, which is documented to round half up (for positive results, which are all that matter): https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-muldiv

Reply:
Somehow I knew what I’d see before I clicked the link 😃

I may have forgotten about that when I wrote this commit. But despite the VSFilter comment, I think there is actually no difference between its two computations: the divisor is always `0xff` which is an odd number, so the quotient simply can’t be exactly halfway two integers and hence is always rounded to the unique closest integer. A quick, _non_-exhaustive practical test failed to find any counterexamples, too.